### PR TITLE
Changed resolvevideos parameter teminology for better clarity

### PR DESCRIPTION
**Pull request by tukeyclothespin**
- directFeatures now processes images in a folder
- resolveVideos creates a filename/contents hash table in /reference
- directFeatures output includes face_pic and frame_pic for demo reference
- updated directFeatures_tinyface to align with changes in directFeatures
- Using /out folder as default location to search for detected faces given readme instructions for directFeatures
- Creating /reference folder as the default location to keep reference set of faces after resolving detected videos
- Updated readme

### CODE REVIEW

File: directFeatures.py [hunk 1]
```diff
@@ -52,11 +50,16 @@ def match_to_faces(list_face_encodings, list_face_locations, people, resized_ima
             next_unknown += 1
             current = people[name]
             current['face_vec'] = face_encoding
-            current['video_name'] = filename
-            current['video_hash'] = file_hash
+            current['file_name'] = filename
+            current['file_name_hash'] = hashlib.md5(str(filename).encode('utf-8')).hexdigest()
```
Comment (d-grossman) [inline directFeatures.py]: hoist the hashlib.md5(...)
above line 29 as it does not need to be computed for every face, only the 1st face.

File: directFeatures.py [hunk 2]
```diff
@@ -69,16 +72,13 @@ def match_to_faces(list_face_encodings, list_face_locations, people, resized_ima
         list_face_names.append(name)
 
         for (top, right, bottom, left), name in zip(list_face_locations, list_face_names):
-            cv2.rectangle(resized_image, (left, top),
-                          (right, bottom), (255, 0, 0), 2)
+            cv2.rectangle(resized_image, (left-5, top-5),
+                          (right+5, bottom+5), (255, 0, 0), 2)
```
Comment (d-grossman) [inline directFeatures.py]: need to check bounds on boxes..
left,top need to be > 5
right,bottom need to be < 5 away from the rt bottom corner of the image

File: resolveVideos.py [hunk 3]
```diff
@@ -39,19 +39,33 @@ def main(detected_faces_folder,reference_faces_file, tolerance):
...
-                                                                                                           '.pickle']]
+    # Load file name/content hash table if available
+    if os.path.isfile(in_hash_table):
+        hash_table = pickle.load(open(in_hash_table, "rb"))
+        print("Loaded hash table from {0} with {1} entries".format(in_hash_table, len(hash_table['hash_to_file'])))
+    else:
+        print("No hash table pickle file found at {0}, starting with blank table".format(in_hash_table))
+        hash_table = defaultdict(dict)
+
+    filtered_folder = [file for file in os.listdir(detected_faces_folder) if file[-20:] == "face_detected.pickle"]
```
Comment (d-grossman) [inline resolveVideos.py]: probably want to do
[file for file in os.listdir(...) if file.endswith('face_detected.pickle')]

File: tinyface/directFeatures_tinyface.py [hunk 4]
```diff
@@ -70,16 +73,13 @@ def match_to_faces(list_face_encodings, list_face_locations, people, resized_ima
         list_face_names.append(name)
 
         for (top, right, bottom, left), name in zip(list_face_locations, list_face_names):
-            cv2.rectangle(resized_image, (left, top),
-                          (right, bottom), (255, 0, 0), 2)
+            cv2.rectangle(resized_image, (left-5, top-5),
+                          (right+5, bottom+5), (255, 0, 0), 2)
```
Comment (d-grossman) [inline tinyface/directFeatures_tinyface.py]: same comment as above, make sure the left,top,rigtht,bottom are still within the image when you add/subtract 5 from them.

**Review by d-grossman — CHANGES_REQUESTED** [review]
small things.

File: resolveVideos.py (hunk 3 above)
Comment (tukeyclothespin) [reply resolveVideos.py]: Didn't know this was possible. nifty.

File: directFeatures.py (hunk 1 above)
Comment (tukeyclothespin) [reply directFeatures.py]: agreed, moved to line 29

File: directFeatures.py (hunk 2 above)
Comment (dave-lab41) [inline directFeatures.py]: based on interpreter testing, cv2 accepts out of bounds coordinates without error and draws partial box as desired

File: tinyface/directFeatures_tinyface.py (hunk 4 above)
Comment (dave-lab41) [inline tinyface/directFeatures_tinyface.py]: based on interpreter testing, cv2 accepts out of bounds coordinates without error and draws partial box as desired

File: directFeatures.py (hunk 2 above)
Comment (d-grossman) [inline directFeatures.py]: acceptable!

File: tinyface/directFeatures_tinyface.py (hunk 4 above)
Comment (d-grossman) [inline tinyface/directFeatures_tinyface.py]: acceptable

**Review by d-grossman — APPROVED** [review]

**Pull request merged**